sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -135,6 +135,18 @@ abstract class JdbcDialect extends Serializable with Logging {
     s""""$colName""""
   }
 
+  /**
+   * Get the SQL query that should be used to create a table. Dialects can
+   * override this method to return a query that works best in a particular database.
+   * @param tableName The name of the table.
+   * @param strSchema The string of the schema.
+   * @param createTableOptions The string of the options that to allow certain options to append
+   *             when create a new table, which can be table_options or partition_options.
+   */
+  def createTable(tableName: String, strSchema: String, createTableOptions: String): String = {
```

Comment:
It seems strSchema prevent supporting different databases if schema is different.
Could you think about more general scenarios ?
I guess we should add another API to customize the schema str by dialect.

Reply:
Yes, you are right, strSchema comes from the existing JdbcUtils.schemaString, this method is to generate schema strings according to different databases

Reply:
Detailed comments have been added for the strSchema parameter. What do you think of this? Please help to review again, thank you @beliefer